tests/test_downloader.py

```diff
@@ -501,6 +501,14 @@ def parse(self, response: DummyResponse):
     assert not isinstance(collected["response"], DummyResponse)
 
 
+def test_dummy_response_multiple_args() -> None:
+    dummy = DummyResponse(url="https://google.com", request=None, status=300)
+    assert dummy.status == 300
+
+    dummy = DummyResponse("https://google.com", 400)
+    assert dummy.status == 400
```

Comment:
Hey! The original issue was that .replace doesn't work; could you please add a test for this, or maybe even change this test to check .replace? 

Users shouldn't be creating DummyResponse instances themselves, but they might have code which calls .replace.